### PR TITLE
[ENH] pull in warning for dialect change from pandas-gbq

### PR DESCRIPTION
**Pull request by tswast**
- [x] ~closes~ towards https://github.com/pydata/pandas-gbq/issues/195
- [x] tests added / passed (N/A)
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/gbq.py [hunk 1]
```diff
@@ -108,6 +112,17 @@ def read_gbq(query, project_id=None, index_col=None, col_order=None,
...
+    if dialect is None:
+        dialect = "legacy"
+        warnings.warn(
+            'The default value for dialect is changing to "standard" in a '
+            'future version of pandas-gbq. Pass in dialect="legacy" to '
+            "disable this warning.",
+            FutureWarning,
+            stacklevel=2,
+        )
+
```
Comment (gfyoung) [inline pandas/io/gbq.py]: Do we have a good way to test this by any chance?

Comment (tswast) [reply pandas/io/gbq.py]: I've got a test in the pandas-gbq repo:

https://github.com/pydata/pandas-gbq/blob/998b6cfe953613bf5ada6253a7d3550837a2d169/tests/unit/test_gbq.py#L367-L369

I suppose I should include a similar test here?

Comment (gfyoung) [reply pandas/io/gbq.py]: If it's possible to include in some shape or form, I would definitely be 👍 for it.

Comment (tswast) [reply pandas/io/gbq.py]: Added in https://github.com/pandas-dev/pandas/pull/22557/commits/07c2eb80875f73fbed8a38b84cd0d5f34246868c

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/test_gbq.py [hunk 2]
```diff
@@ -93,6 +98,14 @@ def make_mixed_dataframe_v2(test_size):
                      index=range(test_size))
 
 
+def test_read_gbq_without_dialect_warns_future_change(monkeypatch):
+    mock_read_gbq = mock.Mock()
+    mock_read_gbq.return_value = DataFrame([[1.0]])
+    monkeypatch.setattr(pandas_gbq, 'read_gbq', mock_read_gbq)
+    with pytest.warns(FutureWarning):
```
[the remark this answers is not in the document]
Comment (tswast) [reply pandas/tests/io/test_gbq.py]: Done in https://github.com/pandas-dev/pandas/pull/22557/commits/4029d97f553246aacdfbed7effe30b75b131606d

File: pandas/tests/io/test_gbq.py [hunk 3]
```diff
@@ -93,6 +98,14 @@ def make_mixed_dataframe_v2(test_size):
                      index=range(test_size))
 
 
+def test_read_gbq_without_dialect_warns_future_change(monkeypatch):
+    mock_read_gbq = mock.Mock()
```
[the remark this answers is not in the document]
Comment (tswast) [reply pandas/tests/io/test_gbq.py]: Done in https://github.com/pandas-dev/pandas/pull/22557/commits/0de8d221c8959f92ee1ae463b6ef553e2a97fc75

**Comment by tswast** [comment]
Travis error appears to be a temporary issue with Conda downloads. https://travis-ci.org/pandas-dev/pandas/jobs/427334975

> CondaHTTPError: HTTP 504 GATEWAY TIME-OUT for url <https://conda.anaconda.org/conda-forge/linux-64/pycryptodome-3.6.6-py27h16a7912_0.tar.bz2>

Maybe retrying it would help?

**Review by jreback — CHANGES_REQUESTED** [review]
tiny comment. ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @tswast

**Pull request merged**